tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -966,6 +966,301 @@ namespace tuplex {
             return SerializableValue(resVal, resSize);
         }
 
+        codegen::SerializableValue FunctionRegistry::createMathIsNanCall(llvm::IRBuilder<>& builder, const python::Type &argsType,
+                                                     const python::Type &retType,
+                                                     const std::vector<tuplex::codegen::SerializableValue> &args) {
+            using namespace llvm;
+            auto& context = builder.GetInsertBlock()->getContext();
+            auto val = args.front();
+            auto type = argsType.parameters().front();
+            
+            if (python::Type::F64 == type) {
+                llvm::Value* i64Val = builder.CreateBitCast(val.val, llvm::Type::getInt64Ty(context));
+                _env.printValue(builder, val.val, "double/python float value\n");
+                
+                auto shiftedVal = builder.CreateLShr(i64Val, 32);
+                auto i32Shift = builder.CreateTrunc(shiftedVal, llvm::Type::getInt32Ty(context));
+                auto andRes = builder.CreateAnd(i32Shift, 2147483647);
+                auto i32Val = builder.CreateTrunc(i64Val, llvm::Type::getInt32Ty(context));
+                auto cmpRes = builder.CreateICmpNE(i32Val, ConstantInt::get(i32Val->getType(), 0));
+                auto i32cmp = builder.CreateZExt(cmpRes, llvm::Type::getInt32Ty(context));
+                auto added = builder.CreateNUWAdd(andRes, i32cmp);
+                auto addCmp = builder.CreateICmpUGT(added, ConstantInt::get(i32Val->getType(), 2146435072));
+
+                auto resVal = _env.upcastToBoolean(builder, addCmp);
+                auto resSize = _env.i64Const(sizeof(int64_t));
+
+                return SerializableValue(resVal, resSize);
+            } else {
+                // only other valid input types are integer and boolean
+                assert(python::Type::BOOLEAN == type || python::Type::I64 == type);
+                _env.printValue(builder, val.val, "boolean or integer value\n");
+                
+                // return SerializableValue(ConstantInt::get(llvm::Type::getInt64Ty(context), 0), _env.i64Const(sizeof(int64_t)));
+                return SerializableValue(_env.boolConst(false), _env.i64Const(sizeof(int64_t)));
+            }
+        }
+
+        codegen::SerializableValue FunctionRegistry::createMathIsInfCall(llvm::IRBuilder<>& builder, const python::Type &argsType,
+                                                     const python::Type &retType,
+                                                     const std::vector<tuplex::codegen::SerializableValue> &args) {
+            using namespace llvm;
+            auto& context = builder.GetInsertBlock()->getContext();
+            auto val = args.front();
+            auto type = argsType.parameters().front();
+
+            if (python::Type::F64 == type) {
+                _env.printValue(builder, val.val, "double/python float value\n");
```

Comment:
remove debug print, add explanation for constants/magic numbers and what the idea here is